keps/sig-apps/20190318-Graduate-CronJob-to-Stable.md

```diff
@@ -0,0 +1,552 @@
+- FailedJobsHistoryLimit
+  - should delete failed finished jobs when above failedJobsHistoryLimit
+- Events Recorder
+  - should not emit unexpected warnings
+  - should remove from active list jobs that have been deleted
+
+#### New test cases
+- Schedule *
+  - Should not create a cronjob with invalid schedule format
+- StartingDeadlineSeconds *
+  - Should not schedule a job within two minutes when missed the current window if 
+  - StartingDeadlineSeconds is 0
+  - Should schedule a job soon when missed the current window if StartingDeadlineSeconds is long 
+- JobTemplate
+  - Should not schedule a job with invalid job template
+- [Endpoints coverage](https://apisnoop.cncf.io/?zoomed=category-beta-batch)
+  - Should list cronjobs for all namespaces
+  - Should update a cronjob
+  - Should patch a cronjob
+  - Should delete all cronjobs in a namespace
+  - Should get a cronjob status
+  - Should update a cronjob status
+  - Should patch a cronjob status
+- Tests covering Bug fixes
+  - [issue/63371 - Should not start “missed” jobs from old cronjob after updating time](https://github.com/kubernetes/kubernetes/issues/63371) 
+  - [issue/74741 - Should cleanup finished pods when job is completed](https://github.com/kubernetes/kubernetes/issues/74741)
+  - [issue/76570 - Should not keep creating new pods when job image has ImagePullBackOff error](https://github.com/kubernetes/kubernetes/issues/76570)
+  - [issue/78646 - Should change shedule when schedule value is not wrapped with quotes](https://github.com/kubernetes/kubernetes/issues/78646)
+- Scaling
+  - Should be able to create and schedule atleast 5000 cronjobs
+- Start Stop Tests
+  - Schedule cronjobs and randomly stop the controller and start it.
+  - Schedule cronjobs and stop the controller and start it after the deadline.
+
+
+### Conformance Tests
+The conformance tests are a subset of e2e tests. We will select test scenarios that we believe are expected from all conforming clusters. Then modify the test case to use the `framework.ConformanceIt()` function rather than the `framework.It()` function. 
+
+
+### Unit Tests
+This is subject to the new rearchitected controller implementation. Overall these scenarios would be tested.
+
+- [Run or Not](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controller_test.go#L167) tests the controller under different scenarios to check if the Job is created or not
+- [Validates Job cleanup](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controller_test.go#L371) path of the controller under different conditions.
+- [Validates Status](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controller_test.go#L593) of the CronJob after sync under different conditions.
+
+
+## Graduation Criteria
+
+- [ ] Implement shared informers to reduce pressure on API Server
+- [ ] Add conformance tests
```

Comment:
I'd say, promote existing tests covering the basic functionality of cronjobs. Maybe it's worth picking that subset inside the conformance section above?

Reply:
Called out a subset.